File: language/transaction-builder/generator/examples/python3/custom_diem_code/AccountAddress.py
# Copyright (c) The Diem Core Contributors
# SPDX-License-Identifier: Apache-2.0

LENGTH = 16  # type: int

def to_bytes(self) -> bytes:
    """Convert account address to bytes."""
    return bytes(typing.cast(typing.Iterable[int], self.value))

@staticmethod
def from_bytes(addr: bytes) -> "AccountAddress":
    """Create an account address from bytes."""
    if len(addr) != AccountAddress.LENGTH:
        raise ValueError("Incorrect length for an account address")
    return AccountAddress(value=tuple(st.uint8(x) for x in addr))  # pyre-ignore

def to_hex(self) -> str:
    """Convert account address to an hexadecimal string."""
    return self.to_bytes().hex()

@staticmethod
def from_hex(addr: str) -> "AccountAddress":
    """Create an account address from an hexadecimal string."""
    return AccountAddress.from_bytes(bytes.fromhex(addr))

File: language/transaction-builder/generator/examples/csharp/custom_diem_code/AccountAddress.cs
// Copyright (c) The Diem Core Contributors
// SPDX-License-Identifier: Apache-2.0

const int LENGTH = 16;

public static AccountAddress valueOf(byte[] values) {
    if (values.Length != LENGTH) {
        throw new ArgumentException("Invalid length for AccountAddress");
    }
    List<byte> address = new List<byte>(LENGTH);
    for (int i = 0; i < LENGTH; i++) {
        address.Add(values[i]);
    }
    return new AccountAddress(new Serde.ValueArray<byte>(address.ToArray()));
}

public byte[] toBytes() {
    byte[] bytes = new byte[LENGTH];
    int i = 0;
    foreach (byte item in value) {
        bytes[i++] = item;
    }
    return bytes;
}
